wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -265,18 +269,20 @@ public void run() {
     loopCache.poll();
     m_watchdog.addEpoch("buttons.run()");
 
-    m_inRunLoop = true;
+    m_deferCancel = true;
+    m_deferSchedule = true;
+    boolean isDisabled = RobotState.isDisabled();
     // Run scheduled commands, remove finished commands.
     for (Iterator<Command> iterator = m_scheduledCommands.iterator(); iterator.hasNext(); ) {
       Command command = iterator.next();
 
-      if (!command.runsWhenDisabled() && RobotState.isDisabled()) {
+      if (!command.runsWhenDisabled() && isDisabled) {
         command.end(true);
         for (Consumer<Command> action : m_interruptActions) {
           action.accept(command);
         }
-        m_requirements.keySet().removeAll(command.getRequirements());
         iterator.remove();
+        m_requirements.keySet().removeAll(command.getRequirements());
         m_watchdog.addEpoch(command.getName() + ".end(true)");
```

Comment:
Isn't this duplicate of `cancelCommand`?

Reply:
Not exactly, since we need to remove the command through the iterator to avoid CME- We could potentially unify them with a lambda, though.